File: JULY-2025/03-07-2025/p3304.scala
object Solution {
  def kthCharacter(k: Int): Char = {
    var asc = List(0)
    while (asc.length < k) {
      val current = asc
      current.foreach { v =>
        if (asc.length < k)
          asc = asc :+ ((v + 1) % 26)
      }
    }
    (asc(k - 1) + 'a'.toInt).toChar
  }
}

File: JULY-2025/03-07-2025/p3304.kt
class Solution {
    fun kthCharacter(k: Int): Char {
        var bitCount = 0
        var k = k - 1
        while (k != 0) {
            bitCount += k and 1
            k = k shr 1
        }
        return 'a' + bitCount
    }
}